exchanges/ftx/ftx_websocket.go

```diff
@@ -0,0 +1,292 @@
+			if err != nil {
+				return err
+			}
+		}
+		switch result["channel"] {
+		case "ticker":
+			var resultData WsTickerDataStore
+			err = json.Unmarshal(respRaw, &resultData)
+			if err != nil {
+				return err
+			}
+			f.Websocket.DataHandler <- &ticker.Price{
+				ExchangeName: f.Name,
+				Bid:          resultData.Ticker.Bid,
+				Ask:          resultData.Ticker.Ask,
+				Last:         resultData.Ticker.Last,
+				LastUpdated:  timestampFromFloat64(resultData.Ticker.Time),
+				Pair:         p,
+				AssetType:    a,
+			}
+		case "orderbook":
+			var resultData WsOrderbookDataStore
+			err = json.Unmarshal(respRaw, &resultData)
+			if err != nil {
+				return err
+			}
+			var newOB orderbook.Base
+			if len(resultData.OBData.Asks) != 0 {
+				for x := range resultData.OBData.Asks {
+					newOB.Asks = append(newOB.Asks, orderbook.Item{Price: resultData.OBData.Asks[x][0],
+						Amount: resultData.OBData.Asks[x][1],
+					})
+				}
+			}
+			if len(resultData.OBData.Bids) != 0 {
+				for y := range resultData.OBData.Bids {
+					newOB.Bids = append(newOB.Bids, orderbook.Item{Price: resultData.OBData.Bids[y][0],
+						Amount: resultData.OBData.Bids[y][1],
+					})
+				}
+			}
+			newOB.Pair = p
+			newOB.AssetType = a
+			newOB.ExchangeName = f.Name
+			err = f.Websocket.Orderbook.Update(&wsorderbook.WebsocketOrderbookUpdate{
+				Asset: newOB.AssetType,
+				Bids:  newOB.Bids,
+				Asks:  newOB.Asks,
+				Pair:  p,
+			})
+			if err != nil {
+				return err
+			}
+			f.Websocket.DataHandler <- wshandler.WebsocketOrderbookUpdate{
+				Pair:     p,
+				Asset:    a,
+				Exchange: f.Name,
+			}
+		case "trades":
+			var resultData WsTradeDataStore
+			err = json.Unmarshal(respRaw, &resultData)
+			if err != nil {
+				return err
+			}
+			for z := range resultData.TradeData {
+				var oSide order.Side
+				oSide, err = order.StringToOrderSide(resultData.TradeData[z].Side)
+				if err != nil {
+					f.Websocket.DataHandler <- order.ClassificationError{
+						Exchange: f.Name,
+						Err:      err,
+					}
+				}
+				f.Websocket.DataHandler <- wshandler.TradeData{Timestamp: resultData.TradeData[z].Time,
+					CurrencyPair: p,
+					AssetType:    a,
+					Exchange:     f.Name,
+					Price:        resultData.TradeData[z].Price,
+					Amount:       resultData.TradeData[z].Size,
+					Side:         oSide,
+				}
+			}
+		case "orders":
+			var resultData WsOrderDataStore
+			err = json.Unmarshal(respRaw, &resultData)
+			if err != nil {
+				return err
+			}
+			pair := currency.NewPairFromString(resultData.OrderData.Market)
+			var assetType asset.Item
+			assetType, err = f.GetPairAssetType(pair)
+			if err != nil {
+				return err
+			}
+			var oSide order.Side
+			oSide, err = order.StringToOrderSide(resultData.OrderData.Side)
+			if err != nil {
+				f.Websocket.DataHandler <- order.ClassificationError{
+					Exchange: f.Name,
+					Err:      err,
+				}
+			}
+			f.Websocket.DataHandler <- wshandler.TradeData{CurrencyPair: pair,
```

Comment:
Orders websocket endpoint isn't trade data, it's our actual orders being passed in.
You will need to populate an `&order.Detail` struct to send to the datahandler filling as much as possible

Reply:
Thanks for making that change! Only thing now is to make it an `&order.Detail`, not just `order.Detail`